presto-product-tests/src/main/java/com/facebook/presto/tests/sqlserver/Select.java

```diff
@@ -0,0 +1,102 @@
+
+import java.sql.SQLException;
+
+import static com.facebook.presto.tests.TestGroups.SQL_SERVER;
+import static com.facebook.presto.tests.TpchTableResults.PRESTO_NATION_RESULT;
+import static com.facebook.presto.tests.sqlserver.SqlServerTpchTableDefinitions.NATION;
+import static com.facebook.presto.tests.sqlserver.TestConstants.CONNECTOR_NAME;
+import static com.facebook.presto.tests.sqlserver.TestConstants.KEY_SPACE;
+import static com.facebook.presto.tests.utils.QueryExecutors.onPresto;
+import static com.teradata.tempto.Requirements.compose;
+import static com.teradata.tempto.assertions.QueryAssert.Row.row;
+import static com.teradata.tempto.assertions.QueryAssert.assertThat;
+import static com.teradata.tempto.fulfillment.table.TableRequirements.immutableTable;
+import static java.lang.String.format;
+
+public class Select
+        extends ProductTest
+        implements RequirementsProvider
+{
+    @Override
+    public Requirement getRequirements(Configuration configuration)
+    {
+        return compose(
+                immutableTable(NATION)
+        );
+    }
+
+    @Test(groups = SQL_SERVER)
+    public void testSelectNation()
+            throws SQLException
+    {
+        String sql = format(
+                "SELECT n_nationkey, n_name, n_regionkey, n_comment FROM %s.%s.%s",
+                CONNECTOR_NAME,
+                KEY_SPACE,
+                NATION.getName());
+        QueryResult queryResult = onPresto()
+                .executeQuery(sql);
+
+        assertThat(queryResult).matches(PRESTO_NATION_RESULT);
+    }
+
+    @Test(groups = SQL_SERVER)
+    public void testNationJoinNation()
+            throws SQLException
+    {
+        String tableName = format("%s.%s.%s", CONNECTOR_NAME, KEY_SPACE, NATION.getName());
+        String sql = format(
+                "SELECT n1.n_name, n2.n_regionkey FROM %s n1 JOIN " +
+                        "%s n2 ON n1.n_nationkey = n2.n_regionkey " +
+                        "WHERE n1.n_nationkey=3",
+                tableName,
+                tableName);
+        QueryResult queryResult = onPresto()
+                .executeQuery(sql);
+
+        assertThat(queryResult).containsOnly(
+                row("CANADA", 3),
+                row("CANADA", 3),
+                row("CANADA", 3),
+                row("CANADA", 3),
+                row("CANADA", 3));
+    }
+
+    @Test(groups = SQL_SERVER)
+    public void testNationJoinRegion()
+            throws SQLException
+    {
+        String sql = format(
+                "SELECT c.n_name, t.name FROM %s.%s.%s c JOIN " +
+                        "tpch.tiny.region t ON c.n_regionkey = t.regionkey " +
+                        "WHERE c.n_nationkey=3",
+                CONNECTOR_NAME,
+                KEY_SPACE,
+                NATION.getName());
+        QueryResult queryResult = onPresto()
+                .executeQuery(sql);
+
+        assertThat(queryResult).containsOnly(row("CANADA", "AMERICA"));
+    }
+}
```

Comment:
Add a test for pushdown (`where` clause).
Also for aggregates (e.g. `count(1)`) because we've seen issues with them previously in other connectors.